pyne/partisn.py

```diff
@@ -0,0 +1,816 @@
+    skip_array = [['mat:Vacuum'], ['mat:vacuum'], ['mat:Graveyard'], ['mat:graveyard']]
+    skip_list = ['mat:Vacuum', 'mat:vacuum', 'mat:Graveyard', 'mat:graveyard']
+    zones_compressed = {}
+    for z, info in zones.iteritems():
+        # check first if the definition is 100% void, keep same if is
+        if zones[z]['mat'] in skip_array and zones[z]['vol_frac'] == [1.0]:
+            zones_compressed[z] = info
+        else:
+            # check for partial void
+            zones_compressed[z] = {'mat':[], 'vol_frac':[]}
+            for i, mat in enumerate(zones[z]['mat']):
+                if mat not in skip_list:
+                    zones_compressed[z]['mat'].append(mat)
+                    zones_compressed[z]['vol_frac'].append(zones[z]['vol_frac'][i])
+    
+    # Eliminate duplicate zones and assign each voxel a zone number.
+    # Assign zone = 0 if vacuum or graveyard and eliminate material definition.
+    voxel_zone = {}
+    zones_mats = {}
+    z = 0
+    match = False
+    first = True    
+    for i, vals in zones_compressed.iteritems():
+        # Find if the zone already exists
+        for zone, info in zones_mats.iteritems():
+            # Iterate through both sets to disregard order
+            match_all = np.empty(len(vals['mat']), dtype=bool)
+            match_all.fill(False)
+            for ii, mat in enumerate(vals['mat']):
+                for jj, mat_info in enumerate(info['mat']):
+                    if mat == mat_info and np.allclose(np.array(vals['vol_frac'][ii]), \
+                                np.array(info['vol_frac'][jj]), rtol=1e-5):
+                        match_all[ii] = True
+                        break
+            if match_all.all() == True:
+                match = True
+                y = zone
+                break
+            else:
+                match = False
+        # Create a new zone if first zone or does not match other zones
+        if first or not match:
+            # Check that the material is not 100% void (assign zone 0 otherwise)
+            if vals['mat'] in skip_array:
+                voxel_zone[i] = 0
+            else:
+                z += 1
+                zones_mats[z] = zones_compressed[i]
+                voxel_zone[i] = z
+                first = False
+        else:
+            if vals['mat'] in skip_array:
+                voxel_zone[i] = 0
+            else:
+                voxel_zone[i] = y
+    
+    # Remove any instances of graveyard or vacuum in zone definitions
+    zones_novoid = {}
+    for z in zones_mats.keys():
+        zones_novoid[z] = {'mat':[], 'vol_frac':[]}
+        for i, mat in enumerate(zones_mats[z]['mat']):
+            if mat not in skip_list:
+                name = strip_mat_name(mat)
+                zones_novoid[z]['mat'].append(name)
+                zones_novoid[z]['vol_frac'].append(zones_mats[z]['vol_frac'][i])
+    
+    # Put zones into format for PARTISN input
+    if 'x' in bounds.keys():
```

Comment:
I guess I never realized that worked. I will have to go through and change that in several places.